google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1beta2/JsonToProtoMessage.java

```diff
@@ -138,7 +139,10 @@ private static void fillField(
         }
         break;
       case BYTES:
-        if (val instanceof String) {
+        if (val instanceof ByteString) {
+          protoMsg.setField(fieldDescriptor, ((ByteString) val).toByteArray());
+          return;
+        } else if (val instanceof String) {
           protoMsg.setField(fieldDescriptor, ((String) val).getBytes());
```

Comment:
If it is a string, we want to decoded it and pass in the byte array. It is well known that bytes needs to be encoded in order to be put into string. I don't think ((String) val).getBytes()) will work for bytes. It needs to be fixed. You can fix it in a separate PR.

Reply:
I will work on that in another PR, I think that this is how it was originally set up and I only added the ByteString version. I think that you're right and it should be changed though. This will require some other changes to make sure that our tests don't break.

Reply:
Got it. SGTM.